src/sentry/tasks/base.py

```diff
@@ -84,10 +86,11 @@ def instrumented_task(name, stat_suffix=None, silo_mode=None, record_timing=Fals
     def wrapped(func):
         @wraps(func)
         def _wrapped(*args, **kwargs):
-            record_queue_wait_time = record_timing
-
-            # Use a try/catch here to contain the blast radius of an exception being unhandled through the options lib
-            # Unhandled exception could cause all tasks to be effected and not work
+            do_record_timing_rollout = False
+            record_timing_rollout = options.get("sentry.tasks.record.timing.rollout")
+            if record_timing_rollout and record_timing_rollout > random():
```

Comment:
You could use `sentry.features.rollout.in_random_rollout` to handle the comparison with random and handling unset option values.